app/main.py

```diff
@@ -2,5 +2,15 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    results = {}
+
+    def wrapper(*args) -> int:
+        key = (func.__name__, args)
+        if key not in results:
+            print("Calculating new result")
+            results[key] = func(*args)
+        else:
+            print("Getting from cache")
+        return results.get(key)
```

Comment:
In order to simplify code readability, you can swap conditions
```
if key in results:
    ....
    return

....
results[key] = ...
return
```

Note that you actually don't need else statement here because if key is in results, it'll return it and will no reach the code after if statement ;)
You could do the same if you added return statement in if statement, then you would be able to remove else statement